src/Treasure/Tui.hs

```diff
@@ -21,21 +24,34 @@ import qualified Graphics.Vty as V
 
 data MainBox
     = AccountsField Int
+    | TagsField Int
     deriving (Eq, Ord, Show)
 
+type TreasureField e = FormFieldState TreasureChest e MainBox
 
 -- Creates the inout form from the given initial 'TreasureChest'.
 mkForm :: forall e . TreasureChest -> Form TreasureChest e MainBox
-mkForm t = newForm (map makeCheckBox $ zip [0..] $ chestAccounts t) t
+mkForm t@TreasureChest{..} = newForm
+    ( toCheckBoxGroup "Accounts" chestAccountsL AccountsField chestAccounts
+   ++ toCheckBoxGroup "Tags"     chestTagsL     TagsField     chestTags
```

Comment:
This actually a good high-level structure :+1: 

But I think that the next difficulty will be place these two groups on left and right sides instead of on top of each other... `++`-ing lists places them on top of each other and we need to think how to position two groups horizontally...